common/lib/dependabot/dependency_group.rb

```diff
@@ -9,12 +9,21 @@
 
 module Dependabot
   class DependencyGroup
-    attr_reader :name, :rules, :dependencies
+    attr_reader :name, :rules, :dependencies, :handled_dependencies
 
     def initialize(name:, rules:)
       @name = name
       @rules = rules
       @dependencies = []
+      @handled_dependencies = Set.new
+    end
+
+    def add_to_handled(*dependencies)
+      @handled_dependencies += dependencies.map(&:name)
+    end
+
+    def add_all_to_handled
+      @handled_dependencies += dependencies.map(&:name)
     end
```

Comment:
I don't feel strongly, but what if we had a `contains_or_handled?` method that we could call on the groups? Then we don't need to explicitly `add_to_handled` if we already add the dependency to the group?

Reply:
But we only want to consider all the dependencies in the group as handled when there are no semver groups, or if a PR already exists for the group. 

Essentially we're marking which dependencies that are _potentially_ in the group as _actually_ in the group as we proceed through this process.

I'm tempted to rename `group.dependencies` to `group.potentially_grouped_dependencies` to make that clear.